gluten-data/src/main/java/org/apache/spark/sql/vectorized/ColumnarBatchRowUtil.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.vectorized;
+
+import org.apache.gluten.exception.GlutenException;
+
+import java.lang.reflect.Field;
+
+public class ColumnarBatchRowUtil {
```

Comment:
I think we don't need this file... As `ColumnarBatches` is already a utility class of `ColumnarBatch`

We can just do the refection inline in method `transferVectors`

Reply:
Because `ColumnarBatchRow` is protected class in spark3.2, so we must use it in same package

Reply:
If so, Let's move method `transferVectors` to this file then rename it with `ColumnarBatchUtil` or something. To store the reflection code together.

Additionally, we may need to set `rowId` as well.

Reply:
Now I only transfer `ColumnarBatchRow` for Heavy ColumnarBatch, do we need this restriction?

Reply:
I add the rowId but not add test for it because we cannot revoke any ColumnarBatch API to test it. We could only call `ColumnarBatchRow`, but it is protected. Use reflection to test it is too complex.